fix(pagination): support accessible markup

dancormier opened the pull request:
[STACKS-687](https://stackoverflow.atlassian.net/browse/STACKS-687), [A11Y-122](https://stackoverflow.atlassian.net/browse/A11Y-122)

---

This PR introduces support for more accessible markup of the pagination component. Existing implementations using the previous markup should render as they did before.

## Notable changes

- In order for pagination item heights to render as desired with the new markup, this PR adds `display: inline-block` to these items.
- I've added a deprecation notice that recommends consumers migrate to the new markup.
- In addition to modifying the markup used in visual tests, I've added `legacy` markup for child elements using the deprecated structure. I figure this will help us identify any issues that could arise when rendering the deprecated markup.

## To test

- Visit the [pagination component documentation page](https://deploy-preview-1870--stacks.netlify.app/product/components/pagination/)
- Inspect the example pagination element to verify the new structure

dancormier committed:
fix(pagination): support accessible markup

[STACKS-687](https://stackoverflow.atlassian.net/browse/STACKS-687)

dancormier committed:
Remove reference to Stacks 2.6.1 in markup deprecation notice

giamir commented on docs/product/components/pagination.html:
```diff
@@ -29,72 +29,113 @@
-        <span class="v-visible-sr">page</span>
-    </a>
-</div>
+<nav class="s-pagination">
```
This is unlikely to be the only navigation landmark in the page so we might want to add an `aria-label="pagination"` in this example that people will copy/paste to prevent them from introducing another accessibility issue.

dancormier replied:
Good call. I've added `aria-label="pagination"` to example code and `role="presentation"` to rendered examples.

giamir commented on screenshots/Chromium/baseline/s-pagination-dark-legacy.ico:
Oh no, github preview doesn't pick up images with the `.ico` extension. This is a bummer. Netlify really need to sort their issue with that `GIT_LFS_FETCH_INCLUDE` env variable not being honored. 😔

dancormier replied:
Maybe there's an image format that will render. I might spend a moment messing around with the extension to find out.

giamir commented on docs/product/components/pagination.html:
```diff
@@ -29,72 +29,113 @@
 <section class="stacks-section">
     {% header "h2", "Example" %}
+
+    {% tip, "warning", "mb24" %}
```
I don't think this note makes a lot of sense here in isolation since it is unlikely people will remember the old markup. 
Maybe we could have a section called legacy example where we add the deprecation warning and that it will be removed in future releases.

This is what I am suggesting from a content hierarchy perspective:

- Example 
  - New markup example
 
- Legacy Example
  - Deprecation warning message
  - Old markup example 
  
  What do you think?

dancormier replied:
Sounds good to me. I've made the change

giamir reviewed (approved):
Great work @dancormier. I have only left a couple of suggestions but I would be ok to merge the PR even as it is. I will leave it up to you if you think it makes sense to follow up on them.

Thanks for taking care of this a11y bug. 🎉

dancormier committed:
Update docs with review suggestions

Addresses review: https://github.com/StackExchange/Stacks/pull/1870#pullrequestreview-2498703726

dancormier commented:
@giamir thank you for your review. I've made the changes you've suggested and I plan to merge this today. If you see anything you think could use revision, let me know!
